src/checkout.c

```diff
@@ -212,6 +212,10 @@ static bool checkout_is_workdir_modified(
 	if (baseitem->size && wditem->file_size != baseitem->size)
 		return true;
 
+	/* if the workdir item is a directory, it cannot be a modified file */
+	if (S_ISDIR(wditem->mode))
+		return false;
```

Comment:
Shouldn't we consider this as modified if `!S_ISDIR(baseitem->mode)`? E.g. something like

```C
if (S_ISDIR(wditem->mode))
    return !S_ISDIR(baseitem->mode);
```